### ISSUE TITLE
[Bug]: Enabling dirs plugin throws touch errors on startup

### ISSUE THREAD

**Issue by mAAdhaTTah** [Bash-it/bash-it#2143]
### Expected behavior

No errors thrown

### Current behavior

After enabling the dirs plugin, I get errors on this line:

https://github.com/Bash-it/bash-it/blob/379813b6711913e17fd18121d26099bad7277b50/plugins/available/dirs.plugin.bash#L71

The parent directories don't exist and touch doesn't create them, so on shell startup, I get `touch: /Users/{user}/.local/state/bash_it/dirs: No such file or directory` unless I create those directories myself.

### Possible solution

Validate the parent directories exist and create them if they don't with `mkdir -p`.

### Steps to reproduce

1. Install bash-it on a clear Ubuntu or OSX install (had this problem on both).
2. Enable `dirs` plugin.

### Bash-it version

3.0.2

### List of enabled plugins, themes and aliases

dirs

### Bash version

5.1.8

### Operating system and version

OSX and Ubuntu

### bash-it doctor output

```
# How to get: bash-it doctor
```

### Your ~/.bashrc

Technically `.bash_profile`:

```bash
#!/usr/bin/env bash

# Load the shell dotfiles, and then some:
# * ~/.extra can be used for other settings you don’t want to commit.
# Source: simonowndesign/dotfiles
for file in ~/.{path,exports,aliases,functions,apps,extra,managed}; do
	[ -r "$file" ] && source "$file"
done
unset file

# Append to the Bash history file, rather than overwriting it
shopt -s histappend

# Case-insensitive globbing (used in pathname expansion)
shopt -s nocaseglob

# Autocorrect typos in path names when using `cd`
shopt -s cdspell

# Enable some Bash 4 features when possible:
# * `autocd`, e.g. `**/qux` will enter `./foo/bar/baz/qux`
# * Recursive globbing, e.g. `echo **/*.txt`
for option in autocd globstar; do
    shopt -s "$option" 2> /dev/null
done
```

**Comment by davidpfarrell** [comment]
I feel a little bad because I use the dirs plugin and knew about this bug for a few weeks but didn't have time to do anything about it.

Thanks for taking time to open an issue - I have submitted a fix in #2147

**Comment by mAAdhaTTah** [comment]
@davidpfarrell No need to feel bad – I appreciate the plugin & the fix. Thanks!
